src/Microsoft.AspNetCore.Server.IISIntegration/AuthenticationHandler.cs

```diff
@@ -38,7 +38,7 @@ public Task AuthenticateAsync(AuthenticateContext context)
                 {
                     context.Authenticated(User, properties: null,
                         description: Options.AuthenticationDescriptions.FirstOrDefault(descrip =>
-                            string.Equals(User.Identity.AuthenticationType, descrip.AuthenticationScheme, StringComparison.Ordinal))?.Items);
+                            string.Equals(context.AuthenticationScheme, descrip.AuthenticationScheme, StringComparison.Ordinal))?.Items);
```

Comment:
I've just double checked and you are right. When `context.AuthenticationScheme` is "Automatic", the Description will end up being null, however the user is still authenticated.

This behaviour is the same as when `User.Identity.AuthenticationType` returns "Kerberos" under IIS Express (user is authenticated with a null Description), as `IISOptions.AuthenticationDescriptions` only contains "Negotiate" and "NTLM".

To me, an empty description seems preferable to throwing an exception, especially when that exception is only going to be thrown on the server, not dev workstations, making it harder to debug. 

Is there a preferred way of fixing this issue? If so I will be happy to update the MR.

Another solution would be to give the WindowsIdentity a type when it is created in https://github.com/aspnet/IISIntegration/blob/dev/src/Microsoft.AspNetCore.Server.IISIntegration/IISMiddleware.cs#L112 (e.g. `var winIdentity = new WindowsIdentity(handle, "IntegratedWindows");`), but this would mean the above description comparison, as well as `IISOptions.AuthenticationDescriptions` is redundant. This would have the added benefit of anyone calling `User.Identity.AuthenticationType` in user code would get the given string, rather than an exception.